addons/event_sale/models/sale_order.py

```diff
@@ -81,13 +81,25 @@ def _update_registrations(self, confirm=True, cancel_to_draft=False, registratio
             if cancel_to_draft:
                 existing_registrations.filtered(lambda self: self.state == 'cancel').action_set_draft()
 
+            first_registration = registration_data[0]['main_registration_id'] if registration_data and len(registration_data) > 0 else False
             for count in range(int(so_line.product_uom_qty) - len(existing_registrations)):
                 registration_vals = {}
                 if registration_data:
                     registration_vals = registration_data.pop()
                 # TDE CHECK: auto confirmation
                 registration_vals['sale_order_line_id'] = so_line.id
-                Registration.create(registration_vals)
+                if not first_registration:
+                    so_registrations = Registration.search([('sale_order_id', '=', so_line.order_id.id), ('event_id', '=', so_line.event_id.id)])
+                    if so_registrations and so_registrations.main_registration_id:
```

Comment:
`if so_registrations.main_registration_id:` is enough as if not registration returned, the so_registrations object will be empty but still exists and you will anyway be able to test main_registration_id

But this condition is wrong.
`if len(so_registration) > 1:`  so_registrations.main_registration_id.id will crash, you need to take the first of the ids I guess ?? unless main_registration is always the same for each so_registration returned.
could this happen to have different main_registrations ? If so, simply do the following to be safe
```
if so_registrations.main_registration_id.ids:
    so_registrations.main_registration_id.ids[0]
```